src/transformers/models/greaselm/feature_extraction_greaselm.py

```diff
@@ -0,0 +1,739 @@
+
+
+logger = logging.get_logger(__name__)
+
+# soft dependency
+if is_spacy_available():
+    try:
+        import spacy
+        from spacy.matcher import Matcher
+    except OSError:
+        logger.error(
+            "GreaseLM models are not usable since `spacy` can't be loaded. "
+            "Please see https://spacy.io/usage for installation instructions."
+            "In addition to spacy, one needs to install the en_core_web_sm package. "
+            "See https://huggingface.co/spacy/en_core_web_sm for more details"
+        )
+
+
+blacklist = [
+    "-PRON-",
+    "actually",
+    "likely",
+    "possibly",
+    "want",
+    "make",
+    "my",
+    "someone",
+    "sometimes_people",
+    "sometimes",
+    "would",
+    "want_to",
+    "one",
+    "something",
+    "sometimes",
+    "everybody",
+    "somebody",
+    "could",
+    "could_be",
+]
+
+merged_relations = [
+    "antonym",
+    "atlocation",
+    "capableof",
+    "causes",
+    "createdby",
+    "isa",
+    "desires",
+    "hassubevent",
+    "partof",
+    "hascontext",
+    "hasproperty",
+    "madeof",
+    "notcapableof",
+    "notdesires",
+    "receivesaction",
+    "relatedto",
+    "usedfor",
+]
+
+
+class GreaseLMFeatureExtractor(FeatureExtractionMixin):
+    r"""
+    Constructs a GreaseLM feature extractor.
+
+    This feature extractor inherits from [`FeatureExtractionMixin`] which contains most of the main methods. Users
+    should refer to this superclass for more information regarding those methods.
+
+    ['GreaseLMFeatureExtractor'] convert CommonSenseQA or OpenBookQA question-answer example(s) into a batch of graph
+    encodings.
+
+    Args:
+        cpnet_vocab_path (`Union[Path, str]`, defaults to concept.txt):
+            Path to the conceptnet vocabulary file.
+        patterns_path (`Union[Path, str]`, defaults to matcher_patterns.json):
+            Path to the matcher patterns file.
+        pruned_graph_path (`Union[Path, str]`, defaults to conceptnet_en_pruned.pickle.gz):
+            Path to the conceptnet graph file encoded as dict of dicts.
+        score_model (`Union[Path, str]`, defaults to roberta-large):
+            Path to the pretrained model to use for concept scoring.
+        device (`str`, defaults to "cuda"):
+            Device to use for the score model.
+        ctx_node_connects_all (`bool`, defaults to False):
+            Whether to connect all nodes to the context node. False by default
+    """
+
+    model_input_names = [""]
+
+    def __init__(
+        self,
+        cpnet_vocab_path: Union[Path, str] = "concept.txt",
+        pattern_path: Union[Path, str] = "matcher_patterns.json",
+        pruned_graph_path: Union[Path, str] = "conceptnet_en_pruned.pickle.gz",
+        score_model: Union[Path, str] = "roberta-large",
+        device: str = "cuda",
+        cxt_node_connects_all: bool = False,
+        **kwargs
+    ):
+        requires_backends(self, ["spacy", "scatter", "sparse"])
```

Comment:
The feature extractor only depends on SpaCy and PyTorch, not sure why we're checking scatter and sparse here.